### PR TITLE
Add conditional model

### PR DESCRIPTION
**Pull request by ewfuentes**

### CODE REVIEW

File: denoising_diffusion_pytorch/denoising_diffusion_pytorch.py [hunk 1]
```diff
@@ -818,15 +832,28 @@ def p_losses(self, x_start, t, noise = None, offset_noise_strength = None):
         return loss.mean()
 
     def forward(self, img, *args, **kwargs):
+        if self.masked_observations:
+            img, x_obs, x_obs_mask = img
+        else:
+            x_obs = None
+            x_obs_mask = None
```
Comment (siddancha) [inline denoising_diffusion_pytorch/denoising_diffusion_pytorch.py]: Isn't it cleaner to just to this instead:

```python
def forward(self, img, *args, x_obs = None, x_obs_mask = None, **kwargs):
    b, c, h, w, device, img_size, = *img.shape, img.device, self.image_size
    ...
```

similar to the structure of the `self.p_losses()` call below?

**Review by siddancha — COMMENTED** [review]
The main concern is that [diffunc](https://github.com/siddancha/diffunc) uses `denoising_diffusion_pytorch.guided_diffusion` and not `denoising_diffusion_pytorch.denoising_diffusion_pytorch`. Ideally, these changes would be made to `guided_diffusion.py`.

The subclass of `guided_diffusion.py` in `diffunc` implements score functions required for MCMC. So if we want to combine the conditional diffusion model with MCMC, we would need to do so. Otherwise, for now, you can continue using `denoising_diffusion_pytorch.denoising_diffusion_pytorch.py` and this maybe fine.

**Comment by ewfuentes** [comment]
Ahh I see... I'll need to dig into `guided_diffusion`  to understand how it's different. I think it would be interesting to try MCMC with a conditional model, but I think it would still be useful to get this PR in so we can run the evaluation pipeline on just the conditional model.

File: denoising_diffusion_pytorch/denoising_diffusion_pytorch.py (hunk 1 above)
Comment (ewfuentes) [reply denoising_diffusion_pytorch/denoising_diffusion_pytorch.py]: That is fair. I'll make this change.

**Review by siddancha — APPROVED** [review]

**Pull request merged**